app/controllers/glossary_terms_controller.rb

```diff
@@ -0,0 +1,154 @@
+# frozen_string_literal: true
+
+# create and edit Glossary terms
+class GlossaryTermsController < ApplicationController
+  before_action :login_required, except: [
+    :index,
+    :show_past,
+    :show
+  ]
+  before_action :store_location, only: [:index, :show, :destroy]
+  before_action :pass_query_params, only: [:destroy, :show_past]
+
+  # ---------- Actions to Display data (index, show, etc.) ---------------------
+
+  def index
+    @glossary_terms = GlossaryTerm.all.order(:name)
+  end
+
+  def show
+    @glossary_term = GlossaryTerm.find(params[:id].to_s)
+    @canonical_url = glossary_term_url
+    @layout = calc_layout_params
+    @objects = @glossary_term.images
+  end
+
+  # ---------- Actions to Display forms -- (new, edit, etc.) -------------------
+
+  def new
+    @copyright_holder = @user.name
+    @copyright_year = Time.now.utc.year
+    @upload_license_id = @user.license_id
+    @licenses = License.current_names_and_ids(@user.license)
+  end
+
+  def edit
+    @glossary_term = GlossaryTerm.find(params[:id].to_s)
+  end
+
+  # ---------- Actions to Modify data: (create, update, destroy, etc.) ---------
+
+  def create
+    glossary_term = \
+      GlossaryTerm.new(user: @user, name: params[:glossary_term][:name],
+                       description: params[:glossary_term][:description])
+
+    if params[:glossary_term][:upload_image]
+      glossary_term.add_image(process_image(image_args))
+    end
+    glossary_term.save
+    redirect_to(glossary_term_path(glossary_term.id))
+  end
+
+  def update
+    glossary_term = GlossaryTerm.find(params[:id].to_s)
+    glossary_term.attributes = params[:glossary_term].
+                               permit(:name, :description)
+    glossary_term.user = @user
+    glossary_term.save
+    redirect_to(glossary_term_path(glossary_term.id))
+  end
+
+  def destroy
```

Comment:
**Should I create a ticket?
If so, should I reference it in controller code?
Should I remove the destroy action and handle it in a separate PR?**
Lack of a `destroy` action is something I noticed, and hoped to correct, long before starting this PR. But I didn't create a ticket. 
This PR seemed like a great place to implement it, as it's a standard CRUD action.
I may be misunderstanding "a code comment referencing a ticket". We don't usually do that; I can't remember seeing such a comment.(But if that's a good practice, then this would be a good place to start.)

Reply:
If there isn't such a ticket, then it sounds like something worth creating if you aren't going to handle it in this PR.  I don't have a strong opinion about whether this is the right place to do it or if it is likely to have surprising consequences.  A reasonably common thing we do at work when we spot something that we decide is outside the scope of the current ticket is add a comment about it in the code referencing the ticket.  Since in general comments should be fairly rare it helps the work stand out so it gets fixed a bit sooner.  I like it as a practice, but don't feel strongly devoted to it.

Reply:
Thanks for the explanation. That does seem like a good practice, so I'll try to remember to follow it.
I am adding `destroy` in this PR because I want to make GlossaryTermsController my model for normalizing the old controllers, and want to include the all the standard Rails CRUD actions.
(I added some Glossary-related tickets while working on this PR, so I'll reference them in the code.)